app/src/main/java/net/aiscope/gdd_app/ui/sample_preparation/SamplePreparationActivity.kt

```diff
@@ -0,0 +1,110 @@
+package net.aiscope.gdd_app.ui.sample_preparation
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import com.google.android.material.snackbar.Snackbar
+import dagger.android.AndroidInjection
+import kotlinx.android.synthetic.main.activity_sample_preparation.*
+import kotlinx.android.synthetic.main.toolbar.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import net.aiscope.gdd_app.R
+import net.aiscope.gdd_app.extensions.select
+import net.aiscope.gdd_app.ui.CaptureFlow
+import net.aiscope.gdd_app.ui.attachCaptureFlowToolbar
+import net.aiscope.gdd_app.ui.microscope_quality.MicroscopeQualityActivity
+import net.aiscope.gdd_app.ui.snackbar.CustomSnackbar
+import net.aiscope.gdd_app.ui.snackbar.CustomSnackbarAction
+import javax.inject.Inject
+
+class SamplePreparationActivity : AppCompatActivity(), SamplePreparationView, CaptureFlow {
+
+    private val defaultFormData: SamplePreparationViewStateModel by lazy {
+        SamplePreparationViewStateModel(
+            getString(R.string.spinner_empty_option),
+            usesGiemsa = true,
+            giemsaFP = true,
+            usesPbs = true,
+            usesAlcohol = true,
+            reusesSlides = false
+        )
+    }
+
+    @Inject
+    lateinit var presenter: SamplePreparationPresenter
+
+    private val parentJob = Job()
```

Comment:
`parentJob.cancel()` should be called in `onDestroy()` to cancel any running jobs. But now in Android Lifecycle Runtime KTX there is `LifecycleOwner.lifecycleScope`, so you could use `lifecycleScope` instead of `coroutineScope` and get rid of the fluff :)

Reply:
Great, thanks! :-)